web/src/app/monitors/monitors-list/monitors-list.component.ts

```diff
@@ -57,16 +60,29 @@ export class MonitorsListComponent implements OnInit, OnDestroy {
    * @param uid the uid of monitor which needs to be deleted
    */
   deleteMonitor(monitorUid: string): void {
-    this.monitors = this.monitors.filter((monitor: MonitorModel) => monitor.uid !== monitorUid);
-    this.saveMonitorSubscription = this.monitorService.saveMonitors(this.projectUid, this.monitors)
-      .subscribe(
-        () => this.router.navigate([`/projects/${this.projectUid}/monitors`]),
-        (error: any): any => this.snackBar.open(error.message, undefined, { duration: 5000 })
-      );
+    let dialogConfig: MatDialogConfig = new MatDialogConfig();
+    dialogConfig = {
+      width: '500px',
+      data: {
+        title: 'Delete Monitor',
+        content: 'Are you sure you want to delete?',
+      },
+    };
+    this.dialogRef = this.dialog.open(DialogConfirmationComponent, dialogConfig);
+    this.dialogRef.afterClosed().subscribe((result: boolean) => {
+      if (result === true) {
+        this.monitors = this.monitors.filter((monitor: MonitorModel) => monitor.uid !== monitorUid);
+        this.saveMonitorSubscription = this.monitorService.saveMonitors(this.projectUid, this.monitors)
+          .subscribe(
+            () => this.router.navigate([`/projects/${this.projectUid}/monitors`]),
+            (error: any): any => this.snackBar.open(error.message, undefined, { duration: 5000 })
+          );
 
-    // When delete a monitor , deleting all its pings
-    this.monitorService
-      .deletePingsByMonitor(this.projectUid, monitorUid);
+        // When delete a monitor , deleting all its pings
+        this.monitorService
```

Comment:
I think this should be chained from the observables above with the `saveMonitors`